nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/common/AbstractMQTTProcessor.java

```diff
@@ -125,6 +126,7 @@ public ValidationResult validate(String subject, String input, ValidationContext
             .name("Client ID")
             .description("MQTT client ID to use. If not set, a UUID will be generated.")
             .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
```

Comment:
```suggestion
            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
```

Since the evaluation is done in ``onScheduled``, there is no flow file available and we're not using the attributes to make the expression language evaluation. You can change the scope to use the Variable Registry.

Reply:
Ah that makes sense, I did not know about Variable Registry. Thanks, committed change request